src/metrics/include/metrics/MetricsReporter.hpp

```diff
@@ -22,108 +22,106 @@
+        }
+    };
+
+public:
     /**
      * @param out print a human-readable listing of all
      *            data-points to this ostream.
-     * @param metricsFormat the format to use. Either "csv" or "sys-perf".
-     * @param perm passkey permission (must be friends with metrics::Registry)
+     * @param metricsFormat the format to use. Must be "csv".
      */
-    void report(std::ostream& out,
-                const std::string& metricsFormat,
-                v1::Permission perm = {}) const {
+    template <typename ReporterClockSource = SystemClockSource>
+    void report(std::ostream& out, const std::string& metricsFormat) const {
+        v1::Permission perm;
+
         // should these values come from the registry, and should they be recorded at
         // time of registry-creation?
-        auto systemTime = std::chrono::system_clock::now().time_since_epoch().count();
-        auto metricsTime = _registry->now(perm).time_since_epoch().count();
+        auto systemTime = std::chrono::duration_cast<std::chrono::nanoseconds>(
+                              ReporterClockSource::now().time_since_epoch())
+                              .count();
+        auto metricsTime = std::chrono::duration_cast<std::chrono::nanoseconds>(
+                               _registry->now(perm).time_since_epoch())
+                               .count();
 
         // if this lives more than a hot-second, put the formats into an enum and do this
         // check & throw in the driver/main program
         if (metricsFormat == "csv") {
             return reportCsv(out, systemTime, metricsTime, perm);
-        } else if (metricsFormat == "sys-perf") {
-            return reportSysperf(out, systemTime, metricsTime, perm);
         } else {
             throw std::invalid_argument(std::string("Unknown metrics format ") + metricsFormat);
         }
     }
 
 private:
-    template <class T>
-    class TD;
-
-    void reportSysperf(std::ostream& out,
-                       long long int,
-                       long long int,
-                       const v1::Permission& perm) const {
-
-        // TODO: Followup from TIG-1070, make this report real data; for now just report
-        //       the number of timer data-points that we saw.
-
-        genny::metrics::clock::duration total;
-        for (const auto& [name, timer] : _registry->getTimers(perm)) {
-            auto& vals = timer.getTimeSeries(perm).getVals(perm);
-            for (const auto& [when, dur] : vals) {
-                total += dur;
-            }
-        }
-
-        auto micros = std::chrono::duration_cast<std::chrono::microseconds>(total);
-
-        out << R"({"storageEngine":"wiredTiger", "results":[{
- "name":"dummy_inserts", "workload":"genny_dummy_workload",
- "start":1537815283.968272, "end":1537817860.423682,
- "results":{"4":{"ops_per_sec":)"
-            << micros.count() << R"(, "ops_per_sec_values":[)" << micros.count() << R"(]}}}]}
-)";
-    }
```

Comment:
love to see this gone. I think also need to remove mention of the format in the program-options thing in DefaultDriver.

Reply:
I had grep'd for "sys-perf" - [it doesn't look like there are any mentions of the valid options in the DefaultDriver help text](https://github.com/mongodb/genny/blob/5a8c92e7e1003af6ed099419a6a2fe88f8ca273e/src/driver/src/DefaultDriver.cpp#L197-L199).